### PR TITLE
A start on the documentation

### LINKED ISSUE #16: Write documentation for Read the Docs

### PR DESCRIPTION
**Pull request by warsaw**
This includes a usage guide and a migration guide.  No doctests.  No API docs (yet).

Closes #16

### CODE REVIEW

**Comment by brettcannon** [comment]
Travis ain't happy.

**Comment by warsaw** [comment]
Yep, I just pushed a fix (hopefully).

File: importlib_resources/docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ Welcome to importlib_resources
+================================
+
+``importlib_resources`` is a library which provides for access to *resources*
+in Python packages.  It provides functionality similar to ``pkg_resources``
+`Basic Resource Access`_ API, but without all of the overhead and performance
+problems of ``pkg_resources``.
+
+In our terminology, a *resource* is a file that is located within an
+yimportable `Python package`_.  Resources can live on the file system or a zip
```
Comment (brettcannon) [inline importlib_resources/docs/index.rst]: `yimportable` is not a word. 😉

File: importlib_resources/docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+file; this API supports both such that users don't have to worry about it.
```
Comment (brettcannon) [inline importlib_resources/docs/index.rst]: "Resources can live on the file system, zip file, or any place that a loader supports the appropriate API for reading resources.

File: importlib_resources/docs/migration.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+supported by ``importlib_resources``, so projects that use other features
+(e.g. entry points) will have to find other solutions.
+``importlib_resources`` primarily supports the following `basic resource
+access`_ APIs:
+
+* ``pkg_resources.resource_filename()``
+* ``pkg_resources.resource_stream()``
+* ``pkg_resources.resource_string()``
+
+One thing to keep in mind.  ``pkg_resources`` defines *resources* to include
```
Comment (brettcannon) [inline importlib_resources/docs/migration.rst]: "keep in mind:"

File: importlib_resources/docs/migration.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+directories.  ``importlib_resources`` does not treat directories as
+resources; since only file are allowed as resources, file names in the
+``importlib_resources`` API may *not* include slashes.
```
Comment (brettcannon) [inline importlib_resources/docs/migration.rst]: "include path separators (e.g. slashes)".

File: importlib_resources/docs/migration.rst [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+The equivalent code in ``importlib_resources`` is pretty straightforward::
+
+    with importlib_resources.open('my.package', 'resource.dat') as fp:
+        my_bytes = fp.read()
+
+
+resource_string()
+=================
+
+In Python 2, it returns them as a ``str``.  In Python 3,
```
Comment (brettcannon) [inline importlib_resources/docs/migration.rst]: "it" isn't clearly specified in the opening sentence.

File: importlib_resources/docs/migration.rst [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+``resource_filename()`` is one of the more interesting APIs because it
+guarantees that the return value names a file on the file system.  This means
+that if the resource is in a zip file, ``pkg_resources()`` will extract the
+file and return the name of the temporary file it created.  The problem is
+that ``pkg_resources()`` also *implicitly* cleans up this temporary file,
+without control or its lifetime by the programmer.
+
+``importlib_resources`` takes a different approach.  Its equivalent API is the
+``path()`` function, which returns a context manager.  This means users have
```
Comment (brettcannon) [inline importlib_resources/docs/migration.rst]: Mention that the context manager provides a `Path` object.

File: importlib_resources/docs/using.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+In general you never have to worry whether your package is on the file system
+or in a zip file, as the ``importlib_resources`` APIs hide those details from
+you.  Sometimes though, you need a path to an actual file on the file system.
+For example, some SSL APIs require a certificate file to be specified by a
+real file system path, and C's ``dlopen()`` function also requires a real file
+system path.
+
+To support this, ``importlib_resources`` provides an API that will extract the
+resource from a zip file to a temporary file, and return the file system path
+to this temporary file.  In order to properly clean up this temporary file,
```
Comment (brettcannon) [inline importlib_resources/docs/using.rst]: Mention it's a `Path` object.

**Comment by warsaw** [comment]
Thanks for the review!  I believe I've addressed all the issues.  Once this lands I'll make sure the RTD page is getting updated.

**Pull request merged**